mongodbreactivestreams/src/main/java/site/ycsb/db/SubscriberHelpers.java

```diff
@@ -0,0 +1,167 @@
+package site.ycsb.db;
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import com.mongodb.MongoTimeoutException;
+import org.bson.Document;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+
+/**
+ *  Subscriber helper implementations for the Quick Tour.
+ */
+public final class SubscriberHelpers {
+
+  /**
+   * A Subscriber that stores the publishers results and provides a latch so can block on completion.
+   *
+   * @param <T> The publishers result type
+   */
+  public static class ObservableSubscriber<T> implements Subscriber<T> {
+    private final List<T> received;
+    private final List<Throwable> errors;
+    private final CountDownLatch latch;
+    private volatile Subscription subscription;
+    private volatile boolean completed;
+
+    ObservableSubscriber() {
+      this.received = new ArrayList<T>();
+      this.errors = new ArrayList<Throwable>();
+      this.latch = new CountDownLatch(1);
+    }
+
+    @Override
+    public void onSubscribe(final Subscription s) {
+      subscription = s;
+    }
+
+    @Override
+    public void onNext(final T t) {
+      received.add(t);
+    }
+
+    @Override
+    public void onError(final Throwable t) {
```

Comment:
nit picking but could probably give a fuller name than `t`

Reply:
Removed this redundant file SubscriberHelpers.java